lua/lsp/null-ls/formatters.lua

```diff
@@ -1,29 +1,12 @@
 local M = {}
-local formatters_by_ft = {}
 
 local null_ls = require "null-ls"
 local services = require "lsp.null-ls.services"
 local Log = require "core.log"
 
-local function list_names(formatters, options)
-  options = options or {}
-  local filter = options.filter or "supported"
-
-  return vim.tbl_keys(formatters[filter])
-end
-
 function M.list_supported_names(filetype)
-  if not formatters_by_ft[filetype] then
-    return {}
-  end
-  return list_names(formatters_by_ft[filetype], { filter = "supported" })
-end
-
-function M.list_unsupported_names(filetype)
-  if not formatters_by_ft[filetype] then
-    return {}
-  end
-  return list_names(formatters_by_ft[filetype], { filter = "unsupported" })
+  local registered_providers = services.list_registered_providers_names(filetype)
+  return registered_providers["NULL_LS_FORMATTING"] or {}
```

Comment:
Hard-coded values "NULL_LS_FORMATTING" and "NULL_LS_DIAGNOSTICS" should be local constants